### ISSUE TITLE
ghost breakpoints

### ISSUE THREAD

**Issue by Amosss** [firefox-devtools/vscode-firefox-debug#176]
Hi

I want to debug my quasar app in vscode when it runs in Firefox. This is my launch.json:

```
{
            "name": "Launch localhost",
            "type": "firefox",
            "request": "launch",
            "reAttach": true,
            "firefoxExecutable": "C:\\Users\\user\\Downloads\\Portable\\FirefoxPortable\\App\\firefox64\\Firefox.exe",
            "url": "http://localhost:8080",
            "webRoot": "${workspaceFolder}",
            "pathMappings": [{
                "url": "http://localhost",
                "path": "${workspaceFolder}"
            }]
},
```
I start Firefox from the launch of vscode, it opens up but... it stops on certain, unknown, points that are not there. I cleared all the breakpoints but still it stop in style section (for example). Pressing F5 continues but then again it stops at weird points where no one asked it to.

What am I missing?
A bonus question: is it possible to debug from inside Firefox itself/debugger tab?
Thanks a lot!

**Comment by hbenl** [comment]
That sounds very weird, I have no idea why this could be happening. Do you have a project with which I can reproduce this?

> is it possible to debug from inside Firefox itself/debugger tab?

Not sure if I understand your question. You can use Firefox' built-in debugger, even while debugging with VS Code. When you break on breakpoints that you have set in Firefox' debugger, the result will appear in that debugger but not in VS Code and vice versa.

**Comment by Amosss** [comment]
1. I have my project but I won't share it here :)

2. Where can I find the code to debug inside Firefox debugger? I find the vue file but not the js that can be debugged.

Thanks

**Comment by hbenl** [comment]
Ah, so you have a Vue project, that could explain your issue: Vue doesn't enable sourcemaps by default, you'll have to turn them on by adding this to your `vue.config.js`:
```
  configureWebpack: {
    devtool: 'source-map'
  }
```

**Comment by Amosss** [comment]
Now the stopping seems to be more contextual (in the vue file, around the code that is really executing) but still in places with no break points.

If I can debug from inside Firefox, it's enough for me, if only I could find the files to debug :)

**Comment by hbenl** [comment]
Unfortunately debugging support in Vue isn't very good: I also have a Vue app and I also can't find the typescript sources from the `.vue` files in either Firefox' or Chrome's built-in debuggers. I can use breakpoints in `.vue` files using this extension, but it doesn't always work very well.
I can also reproduce your "ghost breakpoints" issue now: it happens for me when I set a breakpoint and then reload the app in Firefox, even if I deleted the breakpoint in the meantime. I will try to fic this soon.
